Remove implicit conversion operator from RString to const char* (Win32)

sukibaby opened the pull request:
# For PR #827 .

sukibaby commented:
Force-pushed to break out the DirectXHelpers and ErrorStrings changes into individual commits

sukibaby commented:
DirectXHelper and ErrorStrings changes were removed from this PR and moved in with #857

aeubanks commented on src/RageFileDriverDirect.cpp:
```diff
@@ -356,7 +356,7 @@ RageFileObjDirect::~RageFileObjDirect()
 		/* We failed. */
 		int err = GetLastError();
-		const RString error = werr_ssprintf( err, "Error renaming \"%s\" to \"%s\"", sOldPath.c_str(), sNewPath.c_str() );
+		const RString error = werr_ssprintf( err, "Error renaming \"%s\" to \"%s\"", sOldPath.c_str(), sNewPath.c_str() ).c_str();
```
`werr_ssprintf` already returns an `RString`?

sukibaby replied:
Yes my apologies. I should have reverted this when I was decoupling it from my branch where I was replacing vssprintf calls.

sukibaby replied:
sorry for the delay here. Just pushed the updated commit.

sukibaby committed:
Remove implicit conversion operator from RString to const char* (Win32)

follows up on 6a29f651c748cbc6086199bb9e51e3065171665a and ecfcb11a0040b9e55e01cdeafd28a50b5f2ee70f

aeubanks commented on src/arch/MemoryCard/MemoryCardDriverThreaded_Windows.cpp:
```diff
@@ -127,34 +128,34 @@ void MemoryCardDriverThreaded_Windows::GetUSBStorageDevices( std::vector<UsbStor
+			bIsSpecifiedMountPoint |= EqualsNoCase(MEMCARDMAN->m_sMemoryCardOsMountPoint[p].Get(), sDrive.c_str());
 
-		RString sDrivePath = sDrive + "\\";
+		RString sDrivePath = sDrive.c_str() + RString("\\");
```
is this required?

sukibaby replied:
if we don't need to c_str sDrive then no

aeubanks commented on src/arch/MemoryCard/MemoryCardDriverThreaded_Windows.cpp:
```diff
@@ -114,9 +114,10 @@ void MemoryCardDriverThreaded_Windows::GetUSBStorageDevices( std::vector<UsbStor
 		RString sDrive = ssprintf( "%c:", 'A'+i%26 );
 
-		LOG->Trace( sDrive );
+		LOG->Trace( sDrive.c_str() );
```
I thought we didn't need this

aeubanks reviewed (commented):
mostly lgtm, the changes to calls to win apis and C stdlib are expected

teejusb commented:
also lgtm, after aeubanks comments addressed

sukibaby commented:
Should be ready now.
